polyglot/piranha/README.md

```diff
@@ -1,20 +1,171 @@
+      - [Example 1 (Stale Feature Flag Cleanup)](#example-1-stale-feature-flag-cleanup)
+      - [Example 2 (Structural Find/Replace with built-in cleanup)](#example-2-structural-findreplace-with-built-in-cleanup)
+      - [Example 3 (Structural Find/Replace with custom cleanup)](#example-3-structural-findreplace-with-custom-cleanup)
+  - [Using Polyglot Piranha](#using-polyglot-piranha)
+    - [:snake: Python API](#snake-python-api)
+    - [Installing the Python API](#installing-the-python-api)
+      - [`run_piranha_cli`](#run_piranha_cli)
+        - [Arguments](#arguments)
+        - [Returns](#returns)
+    - [:computer: Command-line Interface](#computer-command-line-interface)
+        - [Get platform-specific binary from releases or build it from source following the below steps:](#get-platform-specific-binary-from-releases-or-build-it-from-source-following-the-below-steps)
+    - [Languages supported](#languages-supported)
+  - [Getting Started](#getting-started)
+    - [Demos](#demos)
+      - [Running the Demos](#running-the-demos)
+        - [Stale Feature Flag Cleanup:](#stale-feature-flag-cleanup)
+        - [Match-only rules:](#match-only-rules)
+        - [Structural Find/Replace](#structural-findreplace)
+        - [Structural Find/Replace with Custom Cleanup](#structural-findreplace-with-custom-cleanup)
+      - [Building upon the *stale feature flag cleanup* demo](#building-upon-the-stale-feature-flag-cleanup-demo)
+  - [*Stale Feature Flag Cleanup* in depth](#stale-feature-flag-cleanup-in-depth)
+    - [Adding support for a new feature flag system](#adding-support-for-a-new-feature-flag-system)
+      - [Adding a new API usage](#adding-a-new-api-usage)
+      - [Parameterizing the behavior of the feature flag API:](#parameterizing-the-behavior-of-the-feature-flag-api)
+    - [Adding Cleanup Rules](#adding-cleanup-rules)
+      - [Example](#example)
+  - [Piranha Arguments :](#piranha-arguments-)
+  - [Contributing](#contributing)
+    - [Naming conventions for the rules](#naming-conventions-for-the-rules)
+    - [Writing tests](#writing-tests)
+
+
+## Overview
+<p style="text-align:center;">
+<img src="images/piranha_architecture.svg" width="800" height="500" alt="Polyglot Piranha Architecture"/>
+</p>
+
+This is the higher level architecture of Polyglot Piranha. 
+At its heart, Polyglot Piranha is a structural find/replacement (rewrite) engine and pre-build language specific cleanup rules like - like simplifying boolean expressions, simplifying `if-else` statements, deleting empty class, deleting files with no type declarations, inline local variables, and many more. 
+A user provides :
+- A set (or, a graph) of structural find/replace rules 
+- Path to the code base
+- Arguments to modify Piranha's behavior (like deleting associated comments)
+When Piranha applies the set (or graph) of user defined rules, it trigger the __pre-built__ language specific cleanup rules to do a deep cleanup. 
```

Comment:
Note that this shows in the same line as above in the rendered readme: https://github.com/uber/piranha/blob/6c8b038ceae810c4fd254fb823be3f761e952405/polyglot/piranha/README.md

Also: "it trigger**s**"